src/Randomizer.Data/Options/ItemSettingOptions.cs

```diff
@@ -0,0 +1,840 @@
+/// early and items that the player should have in their starting inventory
+/// </summary>
+public class ItemSettingOptions
+{
+    /// <summary>
+    /// The UI name of the item/category
+    /// </summary>
+    public string Item { get; set; }
+
+    /// <summary>
+    /// If this is a Metroid item or not
+    /// </summary>
+    public bool IsMetroid { get; set; }
+
+    /// <summary>
+    /// The list of selectable options for this category
+    /// </summary>
+    public List<ItemSettingOption> Options { get; set; }
+
+    /// <summary>
+    /// Constructor for the ItemSettingsOptions
+    /// </summary>
+    /// <param name="item">The UI name for the item/category</param>
+    /// <param name="isMetroid">If this a metroid item or not</param>
+    /// <param name="options">The list of dropdown options for this item/category</param>
+    public ItemSettingOptions(string item, bool isMetroid, params ItemSettingOption[] options)
+    {
+        Item = item;
+        IsMetroid = isMetroid;
+        Options = options.ToList();
+    }
+
+    /// <summary>
+    /// Retrieves what item types the user has requested to start with in their inventory
+    /// </summary>
+    /// <param name="config">The config of the player</param>
+    /// <returns>The item types that the user will start with</returns>
+    public static IEnumerable<ItemType> GetStartingItemTypes(Config config)
+    {
+        var itemOptions = config.ItemOptions;
+        var itemTypes = GetOptions()
+            .Where(x => itemOptions.ContainsKey(x.Item)
+                        && x.Options[itemOptions[x.Item]].MemoryValues != null
+                        && x.Options[itemOptions[x.Item]].MatchingItemTypes != null)
+            .SelectMany(x => x.Options[itemOptions[x.Item]].MatchingItemTypes!)
+            .ToList();
+        return itemTypes;
+    }
+
+    /// <summary>
+    /// Retrieves what items the user has requested to show early
+    /// </summary>
+    /// <param name="config">The config of the player</param>
+    /// <returns>The item types that should show up early</returns>
+    public static IEnumerable<ItemType> GetEarlyItemTypes(Config config)
+    {
+        var itemOptions = config.ItemOptions;
+        var itemTypes = GetOptions()
+            .Where(x => itemOptions.ContainsKey(x.Item)
+                        && x.Options[itemOptions[x.Item]].MemoryValues == null
+                        && x.Options[itemOptions[x.Item]].MatchingItemTypes != null)
+            .SelectMany(x => x.Options[itemOptions[x.Item]].MatchingItemTypes!)
+            .ToList();
+        return itemTypes;
+    }
+
+    /// <summary>
+    /// Retrieves a list of different items/categories and what options are available for them
+    /// </summary>
+    /// <returns></returns>
+    public static IEnumerable<ItemSettingOptions> GetOptions()
+    {
+        return new List<ItemSettingOptions>
+        {
+            new(item: "Bow", isMetroid: false,
```

Comment:
This might be better as an (embedded?) configuration file.

Reply:
No worries! Hope it doesn't seem like I'm commandeering it too much or anything!

Also maybe one day I'll think to run unit tests before I check in changes. Whoops...

As far as the configuration file, I did consider that, but the reason I opted against that was because adding new options would likely require assembly changes to go along with them, especially on the Super Metroid side. So it's not really something _super_ flexible.

Reply:
That being said, I can totally move it to a config file if you'd prefer.

Reply:
I didn't intend for it to be externally configurable, hence the "embedded" part. Just think YAML or JSON is a better format for it.

Reply:
Alrighty, I can definitely make that change.